generate_export_options.rb

```diff
@@ -67,42 +67,79 @@ def export_method(mobileprovision_content)
 
 # Input validation
 options = {
   export_options_path: nil,
-  archive_path: nil
+  archive_path: nil,
+
+  export_method: nil,
+  upload_bitcode: nil,
+  compile_bitcode: nil
 }
 
-parser = OptionParser.new do|opts|
+parser = OptionParser.new do |opts|
   opts.banner = 'Usage: step.rb [options]'
   opts.on('-o', '--export_options_path path', 'Export options path') { |o| options[:export_options_path] = o unless o.to_s == '' }
   opts.on('-a', '--archive_path path', 'Archive path') { |a| options[:archive_path] = a unless a.to_s == '' }
+
+  opts.on('-m', '--export_method string', 'Export method') { |m| options[:export_method] = m unless m.to_s == '' }
+  opts.on('-u', '--upload_bitcode string', 'Upload bitcode') { |u| options[:upload_bitcode] = u unless u.to_s == '' }
+  opts.on('-c', '--compile_bitcode string', 'Recompile from bitcode') { |c| options[:compile_bitcode] = c unless c.to_s == '' }
+
   opts.on('-h', '--help', 'Displays Help') do
     puts opts
     exit
   end
 end
 parser.parse!
 
-log_fail('export_options_path not specified') if options[:export_options_path].to_s == ''
-log_fail('archive_path not specified') if options[:archive_path].to_s == ''
-
 log_info('Configs:')
 log_details("* export_options_path: #{options[:export_options_path]}")
+log_details("* archive_path: #{options[:archive_path]}")
+
+log_details("* export_method: #{options[:export_method]}")
+log_details("* upload_bitcode: #{options[:upload_bitcode]}")
+log_details("* compile_bitcode: #{options[:compile_bitcode]}")
+
+log_fail('export_options_path not specified') if options[:export_options_path].to_s == ''
+
+if options[:export_method].to_s.empty? && options[:archive_path].to_s.empty?
+  log_fail('failed to determin export-method: no archive_path nor export_method provided')
+end
+
+upload_bitcode = ''
+unless options[:upload_bitcode].to_s.empty?
+  if options[:upload_bitcode].casecmp('yes') != 0 && options[:upload_bitcode].casecmp('no') != 0
+    log_faile("invalid upload_bitcode value: #{options[:upload_bitcode]}, valid values: YES, NO")
+  end
+  upload_bitcode = options[:upload_bitcode].upcase
+end
+
+compile_bitcode = ''
+unless options[:compile_bitcode].to_s.empty?
+  if options[:compile_bitcode].casecmp('yes') != 0 && options[:compile_bitcode].casecmp('no') != 0
+    log_faile("invalid compile_bitcode value: #{options[:compile_bitcode]}, valid values: YES, NO")
+  end
+  compile_bitcode = options[:compile_bitcode].upcase
+end
 
 mobileprovision_content = collect_provision_info(options[:archive_path])
-method = export_method(mobileprovision_content)
 
-export_options = {}
-export_options[:method] = method unless method.nil?
+method = options[:export_method]
+method = export_method(mobileprovision_content) if method.to_s.empty?
 
-log_details("* export_options: #{export_options}")
+log_fail('failed to detect export-method or no export_method provided') if method.to_s.empty?
+
+export_options = {}
+export_options[:method] = method unless method.to_s.empty?
```

Comment:
no need to check with `unless` - we fail 3 lines above if it would be empty (as this is a required propery)